src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,56 @@
+package duke;
+
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.ToDoCommand;
+
+/**
+ * This class is responsible for parsing the input provided by the user into the respective commands they represent
+ */
+public class Parser {
+
+    /**
+     * Produces a Command object corresponding to the input provided. Throws an exception if invalid input is provided.
```

Comment:
perhaps you could add an empty line between the description and parameter section as per the java coding standard? (along for the other javadoc comments as well!) :)